apps/api/src/app/modules/user/users.controller.ts

```diff
@@ -34,6 +43,60 @@ export class UsersController extends CrudController(UserDTO)<IUser> {
     });
   }
 
-
+  @HttpCode(HttpStatus.ACCEPTED)
+  @UseGuards(AuthGuard('jwt'))
+  // @UseInterceptors(RequestDataTypeInterceptor)
+  // @UseInterceptors(
+  //   FileInterceptor('photo', {
+  //     storage: new FileStorage().storage({
+  //       dest: () => {
+  //         return join('photo', moment().format('YYYY/MM'));
+  //       },
+  //       prefix: 'photo',
+  //     }),
+  //   })
+  // )
+  @Put('update/profile')
+  async updateProfile(
+    @Body() entity: any,
+    // @UploadedFileStorage() photo
+  ): Promise<User> {
+    // if (photo?.key) {
+    //   entity.photo = photo?.key;
+    //   await resizeImage(entity.photo, {
+    //     height: 200,
+    //   });
+    // }
+    return this.userService.updateProfile(entity);
+  }
+  
   
+  @HttpCode(HttpStatus.ACCEPTED)
+  @Post('change-password')
+  @UseGuards(AuthGuard('jwt'))
+  async changePassword(@Body() entity: ChangePasswordInput) {
+    const currentUser = RequestContext.currentUser();
+    const user = await this.userService.getOne(currentUser.id);
+    const userPassword = await this.userService.userRepository
+      .createQueryBuilder()
+      .where({
+        id: currentUser.id,
+      })
+      .select('"passwordHash"')
+      .getRawOne();
+
+    const isMatch = await bcrypt.compare(
+      entity.oldPassword,
+      userPassword.passwordHash
+    );
+    if (isMatch) {
+      await this.userService.userRepository.update(user.id, {
+        passwordHash: hashPassword(entity.password)
+      });
+    } else {
+      throw new BadRequestException('Old password is wrong');
+    }
+    return true;
```

Comment:
instead of true should return success message

Reply:
add message